postprocessing/comstockpostproc/comstock_query_builder.py

```diff
@@ -0,0 +1,328 @@
+"""
+Query templates and builders for ComStock data extraction from Athena.
+
+This module provides SQL query templates and builder functions to construct
+Athena queries for various ComStock analysis needs.
+"""
+
+import logging
+from typing import Dict, List, Optional, Union
+
+logger = logging.getLogger(__name__)
+
+
+class ComStockQueryBuilder:
+    """
+    Builder class for creating SQL queries to extract ComStock data from Athena.
+
+    This class provides templated queries that can be customized with parameters
+    for different analysis needs while maintaining consistent query structure.
+    """
+
+    def __init__(self, athena_table_name: str):
+        """
+        Initialize the query builder with the base Athena table name.
+
+        Args:
+            athena_table_name: Base name of the Athena table (without _timeseries or _baseline suffix)
+        """
+        self.athena_table_name = athena_table_name
+        self.timeseries_table = f"{athena_table_name}_timeseries"
+        self.baseline_table = f"{athena_table_name}_baseline"
+
+    #TODO: this method has not yet been validated, and should be prior to use.
+    def get_monthly_energy_consumption_query(self,
+                                           upgrade_ids: Optional[List[Union[int, str]]] = None,
```

Comment:
code formatting request! for multi line function definitions please do it like:
```py
def get_monthly_energy_consumption_query(
    self,
    upgrade_ids: Optional[List[Union[int, str]]] = None,
    states: Optional[List[str]] = None,
    building_types: Optional[List[str]] = None) -> str:
)
```
somewhere there's a pep standard for this but whatever. The big reason to do that is because otherwise you can still end up with wrapping issues should the function input variable name and type definition + the function name exceeds 120 characters.